tests/unit/test_charm.py

```diff
@@ -233,47 +186,26 @@ def test_smtp_integration_container_down(smtp_configured: Harness) -> None:
     assert "Waiting for" in str(harness.model.unit.status)
 
 
-def test_smtp_relation_pebble_success(smtp_configured: Harness, monkeypatch: pytest.MonkeyPatch):
+def test_smtp_relation_success(smtp_configured: Harness, monkeypatch: pytest.MonkeyPatch):
     """
     arrange: start the Synapse charm, set server_name, mock synapse.enable_smtp.
     act: emit smtp_data_available
-    assert: synapse.enable_smtp is called once and unit is active.
+    assert: synapse.enable_smtp is called once.
     """
     harness = smtp_configured
     enable_smtp_mock = MagicMock()
     monkeypatch.setattr(synapse, "enable_smtp", enable_smtp_mock)
-    pull_mock = MagicMock(return_value=io.StringIO("{}"))
-    push_mock = MagicMock()
-    container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
-    monkeypatch.setattr(container, "pull", pull_mock)
-    monkeypatch.setattr(container, "push", push_mock)
+    container: ops.Container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+    container.push(
+        synapse.SYNAPSE_CONFIG_PATH, f'server_name: "{TEST_SERVER_NAME}"', make_dirs=True
+    )
 
     harness.begin()
 
     relation = harness.charm.framework.model.get_relation("smtp", 0)
     harness.charm._smtp.smtp.on.smtp_data_available.emit(relation)
 
     enable_smtp_mock.assert_called_once()
-    assert isinstance(harness.model.unit.status, ops.ActiveStatus)
-
-
-def test_smtp_relation_pebble_error(smtp_configured: Harness, monkeypatch: pytest.MonkeyPatch):
```

Comment:
Why isn't this scenario tested anymore?

Reply:
Same thing than SAML :)
The smtp also called directly the pebble.enable_smtp, not anymore.